### PR TITLE
Refs #925 -- Added build for spidermonkey

### PR DESCRIPTION
**Pull request by alex**
Note that I'm neither on the Firefox JS team nor on the fuzzing team (I work on sandboxing :-)), but this seemed interesting.

Modeled after the Chakra integration -- I have no way of testing besides that `$OUT/js` works, since I don't know how your fuzzer works.

### CODE REVIEW

**Comment by inferno-chromium** [comment]
This needs some manual config and I think Oliver has some CL for this as well. Will let him review.

File: projects/spidermonkey/build.sh [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+cd js/src/
+
+autoconf2.13
+
+mkdir build_DBG.OBJ
+cd build_DBG.OBJ
+
+../configure \
+    --enable-debug \
+    --disable-optimize \
```
Comment (oliverchang) [inline projects/spidermonkey/build.sh]: why disable optimize?

File: projects/spidermonkey/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+################################################################################
+
+FROM gcr.io/oss-fuzz-base/base-builder
+MAINTAINER agaynor@mozilla.com
+RUN apt-get update && apt-get install -y \
+  mercurial \
+  autoconf2.13 \
+  yasm
+
+RUN hg clone https://hg.mozilla.org/mozilla-central/
```
Comment (oliverchang) [inline projects/spidermonkey/Dockerfile]: can we clone from the git repo instead with --depth=1? it seems much faster.

File: projects/spidermonkey/build.sh [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+################################################################################
+
+# Required for some reason... I don't ask questions
+export SHELL=/bin/bash
+cd js/src/
```
Comment (oliverchang) [inline projects/spidermonkey/build.sh]: nit: just set WORKDIR to checkout/js/src in Dockerfile.

File: projects/spidermonkey/Dockerfile (hunk 2 above)
Comment (alex) [reply projects/spidermonkey/Dockerfile]: https://github.com/mozilla/gecko-dev ?

File: projects/spidermonkey/build.sh (hunk 1 above)
Comment (alex) [reply projects/spidermonkey/build.sh]: Good point, no reason to, I think I just got it from the same build instructions. (We do want `enable-debug` for the extra asserts though, I think)

**Comment by alex** [comment]
Sorry about the duplicate work :-)

File: projects/spidermonkey/Dockerfile (hunk 2 above)
Comment (oliverchang) [reply projects/spidermonkey/Dockerfile]: yep!

File: projects/spidermonkey/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+################################################################################
+
+FROM gcr.io/oss-fuzz-base/base-builder
+MAINTAINER agaynor@mozilla.com
+RUN apt-get update && apt-get install -y \
+  autoconf2.13 \
+  yasm
+
+RUN git clone --depth=1 https://github.com/mozilla/gecko-dev mozilla-central
+WORKDIR mozilla-central/js/src/
```
Comment (oliverchang) [inline projects/spidermonkey/Dockerfile]: I think this has to be changerd to gecko-dev/js/src instead.

Comment (alex) [reply projects/spidermonkey/Dockerfile]: Fixed in the next commit, I changed the `git` command to clone to m-c

**Pull request merged**